### PR TITLE
2D-UNet Image Segmentation of HipMRI slices

### PR DESCRIPTION
**Pull request by LiftedFordRanger**
This repository contains a model which attempts to segment 2D slices of HipMRI data using a UNet. 
It contains the following files:
dataset.py - Contains subroutines used for loading and preprocessing data in the nifti format
modules.py - Contains the structure for the UNet model
train.py - The main training script for training the model
predict.py - Used for model inference by loading a set a model weights
graphs - folder containing predicted and true segmentation masks, dice loss during training and images used in the README

Student Number- 46991638

### CODE REVIEW

**Comment by yexincheng** [comment]
**This is an initial inspection, no action is required at this point**
1. **Solves problem**: The problem is appropriately solved, the average DICE score reached around 075. DICE score and prediction figures of validation set are reported.
2. **Implementation**: Hpyerparameters were hard-coded. Training [metric](https://github.com/LiftedFordRanger/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_46991638/train.py#L24) and prediction [metric](https://github.com/LiftedFordRanger/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_46991638/predict.py#L20 ) should be DICE rather than DICE loss (1- DICE). Othwise all good.
3. **Design**: Good.
4. **Comment**: Basic comments, [some functions](https://github.com/LiftedFordRanger/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_46991638/dataset.py#L114-L137) do not have comments. More detailed docstring would be better. A wrong [comment](https://github.com/LiftedFordRanger/PatternAnalysis-2024/blob/topic-recognition/recognition/2D_UNet_46991638/train.py#L32), it is ploting DICE score not loss.
5. **Difficulty**: **Easy**

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. </td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. </td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-13</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Daniel Cottrell

**Comment by LiftedFordRanger** [comment]
Hey team I just noticed I only put my student number not my name I'm not sure who Daniel Cottrell is but I'm Jayke Braga. My student number above is correct 46991638. My grades aren't showing up on blackboard either.

**Comment by shakes76** [comment]
Cant merge because of conflicting changes to main repo files. Please update for merge, doesn't affect grade.
